src/containers/StageMenu.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { actionCreators as stageActions } from '../ducks/modules/stage';
+import { getStages } from '../selectors/session';
+import { Menu } from '../components';
+
+/**
+  * Renders a Menu using stages to construct items in the menu
+  */
+class StageMenu extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchTerm: '',
+      matchingStages: this.props.stages
+    };
+
+    this.onInputChange = this.onInputChange.bind(this);
+  }
+
+  onInputChange(event) {
+    this.setState({
+      searchTerm: event.target.value,
+      matchingStages: this.props.stages.filter(
+        (stage) => stage.title.toLowerCase().includes(event.target.value.toLowerCase()))
+    });
+  }
+
+  render() {
+    let stages = this.props.stages;
+    if (this.state.searchTerm) {
+      stages = this.state.matchingStages;
+    }
+
+    const items = stages.map((stage) =>
+      {
+        return {
+          to: '/protocol/'+stage.id,
+          id: stage.id,
+          title: stage.title,
+          imageType: stage.type,
+          onClick: () => this.props.onStageClick(this.props.stages, stage.id)
+        };
+      }
+    );
+
+    const search = (
+      <div className='bm-search'>
```

Comment:
I think this might be a remnent of existing code, but can we migrate this to use hyphenated bem styles, with a file per component, see some of the other components for examples, but basically:
 
```scss
// general format:

.component__sub-component--variants {
   ...
}
```

Which can be written in scss as:

```scss
// in menu.scss
.menu {
  ...
  &__menu-item {
     ...
     &--active {
       ...
     }
  }
}

```